File: backend/routes/alerts.py
# backend/routes/alerts.py
from fastapi import APIRouter, HTTPException, Body
from typing import List
from models.portfolio_model import PriceAlert
from database import alerts_collection
from bson import ObjectId

router = APIRouter()

@router.post("/")
async def create_alert(alert: PriceAlert):
    """
    Creates a new price alert.
    """
    alert_dict = alert.dict()
    await alerts_collection.insert_one(alert_dict)
    return {"message": f"Alert set for {alert.symbol} {alert.condition} {alert.target_price}"}

@router.get("/{user_id}")
async def get_user_alerts(user_id: str):
    """
    Fetches all active alerts for a user.
    """
    cursor = alerts_collection.find({"user_id": user_id, "status": "ACTIVE"})
    alerts = await cursor.to_list(length=100)
    for a in alerts:
        a["_id"] = str(a["_id"])
    return alerts

@router.delete("/{alert_id}")
async def delete_alert(alert_id: str):
    """
    Deletes an alert.
    """
    result = await alerts_collection.delete_one({"id": alert_id})
    if result.deleted_count == 0:
        raise HTTPException(status_code=404, detail="Alert not found")
    return {"message": "Alert deleted"}

File: frontend/src/api/alerts.js
// src/api/alerts.js
import client from "./client";

export const createAlert = async (userId, symbol, targetPrice, condition) => {
  try {
    const payload = {
        user_id: userId,
        symbol: symbol,
        target_price: parseFloat(targetPrice),
        condition: condition
    };
    const response = await client.post("/alerts/", payload);
    return response.data;
  } catch (error) {
    console.error("Error creating alert:", error);
    throw error.response?.data || error;
  }
};

export const getUserAlerts = async (userId) => {
  try {
    const response = await client.get(`/alerts/${userId}`);
    return response.data;
  } catch (error) {
    console.error("Error fetching alerts:", error);
    throw error.response?.data || error;
  }
};

export const deleteAlert = async (alertId) => {
  try {
    const response = await client.delete(`/alerts/${alertId}`);
    return response.data;
  } catch (error) {
    console.error("Error deleting alert:", error);
    throw error.response?.data || error;
  }
};